src/base/utils/net.cpp

```diff
@@ -38,5 +40,44 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isValidSubnet(const QPair<QHostAddress, int> &subnet)
+        {
+            return subnet != qMakePair(QHostAddress(), -1);
+        }
+
+        bool isLoopbackAddress(const QHostAddress &addr)
+        {
+            return addr == QHostAddress::LocalHost
+                    || addr == QHostAddress::LocalHostIPv6
+                    || addr == QHostAddress(QLatin1String("::ffff:127.0.0.1"));
+        }
+
+        bool isIPInAuthBypassSubnet(const QString &ip, const QList<QPair<QHostAddress, int> > &subnets)
```

Comment:
As this function began to have a more General meaning, please change it name accordingly (remove AuthBypass).